test/test_linalg.py

```diff
@@ -2897,13 +2897,17 @@ def test_svd_errors_and_warnings(self, device, dtype):
                     svd(a, out=(out_u, out_s, out_v))
 
             # if input contains NaN then an error is triggered for svd
+            error_msg = 'The algorithm failed to converge' if self.device_type == 'cpu' \
+                        else 'CUSOLVER_STATUS_EXECUTION_FAILED'
             a = torch.full((3, 3), float('nan'), dtype=dtype, device=device)
             a[0] = float('nan')
-            with self.assertRaisesRegex(RuntimeError, "The algorithm failed to converge"):
+            with self.assertRaisesRegex(RuntimeError, error_msg):
                 svd(a)
+            error_msg = r'\(Batch element 1\): The algorithm failed to converge' if self.device_type == 'cpu' \
+                        else 'CUSOLVER_STATUS_EXECUTION_FAILED'
```

Comment:
I'm not quite certain how to change this test. Cusolver gesvd reports `CUSOLVER_STATUS_EXECUTION_FAILED` when the matrix contains `nan`

Reply:
Could we do something like this?
https://github.com/pytorch/pytorch/issues/64818#issuecomment-917476974

Reply:
> About the error, a cool way to handle this would be to do an input.in_nan().any() before parsing the error message in the function that parses the error 

I don't think this is a good idea. First, `Tensor.isnan()` takes extra time. Besides that, `if (Tensor.isnan().any())` will cause a device-host sync, which further complicates the performance.

Reply:
Note that I propose doing this whenever we know that an error has occured, so we'll be pesimizing the code before throwing an exception.